### PR TITLE
CASH-452: Instantiate Algolia List View

### PR DESCRIPTION
**Pull request by ksekhri**

### CODE REVIEW

File: src/pages/Lend/LendFilterPage.vue [hunk 1]
```diff
@@ -1,81 +1,132 @@
...
+				return Promise.all(cash103);
+			});
+		}
 	},
 };
 </script>
 
 <style lang="scss" scoped>
 @import 'settings';
-@import 'global/transitions';
```
Comment (ksekhri) [inline src/pages/Lend/LendFilterPage.vue]: @mcstover removed `global/transitions` import, were we using this in this page?

Comment (mcstover) [reply src/pages/Lend/LendFilterPage.vue]: yea that's not needed at this point.

File: src/pages/Lend/LendFilterPage.vue [hunk 2]
```diff
@@ -1,81 +1,132 @@
...
+	inject: [
+		'apollo',
+		'algoliaConfig'
+	],
+	apollo: {
+		preFetch(config, client) {
+			return client.query({
+				query: itemsInBasketQuery,
+			}).then(({ data }) => {
+				this.itemsInBasket = _map(_get(data, 'shop.basket.items.values'), 'id');
```
Comment (mcstover) [inline src/pages/Lend/LendFilterPage.vue]: When using the prefetch function you're not in the context of the component so you don't need this here.

In the created hook you'll read from your basket items query and then set this data, There will also need to be a watch query at some point but we can wait on that part and deal with it alongside the adapter component work.

File: src/pages/Lend/LendFilterPage.vue [hunk 3]
```diff
@@ -1,81 +1,132 @@
...
+			}).then(() => {
+				// TODO: REMOVE Once CASH-103: Lend Increment Button EXP ENDS
+				const cash103 = [
+					// Pre-fetch the setting for lend increment button
+					client.query({ query: experimentSetting, variables: { key: 'uiexp.lend_increment_button' } }),
+					// Pre-fetch the assigned version for lend increment button
+					client.query({ query: experimentQuery, variables: { id: 'lend_increment_button' } }),
+				];
```
Comment (mcstover) [inline src/pages/Lend/LendFilterPage.vue]: You can't actually chain these under promise all. The 2nd one requires the 1st so you'll just have to use .then for both. I tried to do this earlier and the experiment assignment would never occur.

Comment (ksekhri) [reply src/pages/Lend/LendFilterPage.vue]: Yes! Good catch hahaha.

File: src/pages/Lend/LendFilterPage.vue [hunk 4]
```diff
@@ -1,81 +1,130 @@
...
+	apollo: {
+		preFetch(config, client) {
+			return client.query({
+				query: itemsInBasketQuery,
+			}).then(() => {
+				// TODO: REMOVE Once CASH-103: Lend Increment Button EXP ENDS
+				return client.query({ query: experimentSetting, variables: { key: 'uiexp.lend_increment_button' } });
+			}).then(() => {
+				// TODO: REMOVE Once CASH-103: Lend Increment Button EXP ENDS
+				return client.query({ query: experimentQuery, variables: { id: 'lend_increment_button' } });
```
Comment (ksekhri) [inline src/pages/Lend/LendFilterPage.vue]: @mcstover cleaned up a bit

File: src/pages/Lend/LendFilterPage.vue [hunk 5]
```diff
@@ -1,81 +1,130 @@
...
+		AlgoliaPagination,
 	},
-
 	metaInfo: {
 		title: 'Lend Filter'
 	},
-
 	created() {
+		const itemsInBasketResults = this.apollo.readQuery({ query: itemsInBasketQuery });
+		this.itemsInBasket = _map(_get(itemsInBasketResults, 'shop.basket.items.values'), 'id');
```
Comment (ksekhri) [inline src/pages/Lend/LendFilterPage.vue]: @mcstover Moved this into created as we discussed

**Pull request merged**